tests/test_filters.py

```diff
@@ -319,3 +320,66 @@ def test_filter_with_non_existing_op(self):
             batch.append(record)
         with self.assertRaisesRegexp(InvalidOperator, 'operator not valid'):
             KeyValueFilter({'options': {'keys': keys}}, meta())
+
+
+class DupeFilterTest(unittest.TestCase):
+
+    def test_filter_duplicates_with_default_key(self):
+        keys = ['8062219f00c79c88', '1859834d918981df', 'e2abb7b480edf910']
+        items = [
+            {'_key': keys[0], 'name': 'item1', 'country_code': 'es'},
+            {'_key': keys[0], 'name': 'item1', 'country_code': 'es'},
+            {'_key': keys[1], 'name': 'item2', 'country_code': 'us'},
+            {'_key': keys[1], 'name': 'item2', 'country_code': 'us'},
+            {'_key': keys[2], 'name': 'item3', 'country_code': 'uk'},
+            {'_key': keys[2], 'name': 'item3', 'country_code': 'uk'}
+        ]
+        batch = []
+        for item in items:
+            record = BaseRecord(item)
+            batch.append(record)
+        filter = DupeFilter({'options': {}}, meta())
+
+        batch = filter.filter_batch(batch)
+        batch = list(batch)
+        self.assertEqual(3, len(batch))
```

Comment:
Just to be a little more sure we're actually getting what we're expecting we can:

- change `keys` to be a set
- `self.assertEqual(keys, set(item['_key'] for item in batch))`

The same can be done in the next test